packages/react/src/useSecurityKeys.ts

```diff
@@ -0,0 +1,151 @@
+import { useEffect, useState } from 'react'
+
+import {
+  ActionErrorState,
+  ActionSuccessState,
+  AddSecurityKeyHandlerResult,
+  addSecurityKeyPromise,
+  ErrorPayload,
+  OTHER_ERROR_CODE
+} from '@nhost/core'
+
+import { useNhostClient } from './useNhostClient'
+
+interface AddSecurityKeyHandler {
+  (
+    /** Optional human-readable name of the security key */
+    nickname?: string
+  ): Promise<AddSecurityKeyHandlerResult>
+}
+interface RemoveSecurityKeyHandlerResult extends ActionErrorState, ActionSuccessState {}
+interface RemoveSecurityKeyHandler {
+  (
+    /** Unique identifier of the security to remove */
+    id: string
+  ): Promise<RemoveSecurityKeyHandlerResult>
+}
+
+export type SecurityKey = { id: string; nickname?: string }
```

Comment:
We should use interfaces wherever possible and wherever it makes sense to be consistent.